src/LockTable.cc

```diff
@@ -0,0 +1,225 @@
+    // index at 1.
+    uint32_t entryIndex = 1;
+    while (true) {
+        for (; entryIndex < ENTRIES_PER_CACHE_LINE; entryIndex++) {
+            if (cacheLine->entries[entryIndex] == lockId) {
+                return true;
+            }
+        }
+        if (cacheLine->next != NULL) {
+            entryIndex = 0;
+            cacheLine = cacheLine->next;
+        } else {
+            break;
+        }
+    }
+    return false;
+}
+
+/**
+ * Releases the lock with the provided lockId.
+ *
+ * \param lockId
+ *      ID of lock to be unlocked.  ID must not be zero.
+ */
+void
+LockTable::releaseLock(uint64_t lockId)
+{
+    // Find the right bucket.
+    uint64_t bucketIndex = (lockId & bucketIndexHashMask);
+    CacheLine* cacheLine = &buckets[bucketIndex];
+
+    // Lock the bucket
+    BucketLock* bucketLock =
+            reinterpret_cast<BucketLock*>(&cacheLine->entries[0]);
+    std::lock_guard<BucketLock> lock(*bucketLock);
+
+    // The zeroth entry in the first CacheLine is the BucketLock so start the
+    // index at 1.
+    uint32_t entryIndex = 1;
+    while (true) {
+        for (; entryIndex < ENTRIES_PER_CACHE_LINE; entryIndex++) {
+            if (cacheLine->entries[entryIndex] == lockId) {
+                cacheLine->entries[entryIndex] = 0;
+                return;
+            }
+        }
+        if (cacheLine->next != NULL) {
+            entryIndex = 0;
+            cacheLine = cacheLine->next;
+        } else {
+            break;
+        }
+    }
+}
+
+/**
+ * Attempts to acquire the lock with the provided lockId without blocking.
+ *
+ * \param lockId
+ *      ID of lock to be acquired.  ID must not be zero.
+ *
+ * \return
+ *      TRUE if the lock was acquired, FALSE otherwise.
+ */
+bool
+LockTable::tryAcquireLock(uint64_t lockId)
+{
+    // Find the right bucket.
+    uint64_t bucketIndex = (lockId & bucketIndexHashMask);
+    CacheLine* cacheLine = &buckets[bucketIndex];
+
+    // Lock the bucket
+    BucketLock* bucketLock =
+            reinterpret_cast<BucketLock*>(&cacheLine->entries[0]);
+    std::lock_guard<BucketLock> lock(*bucketLock);
+
+    int overflowCacheLineCnt = 0;
+
+    // If there is an empty entry slot, store its address here.
+    Entry* entryPtr = NULL;
+
+    // The zeroth entry in the first CacheLine is the BucketLock so start
+    // the index at 1.
+    uint32_t entryIndex = 1;
+    while (true) {
+        for (; entryIndex < ENTRIES_PER_CACHE_LINE; entryIndex++) {
+            if (entryPtr == NULL && cacheLine->entries[entryIndex] == 0) {
+                entryPtr = &cacheLine->entries[entryIndex];
+            } else if (cacheLine->entries[entryIndex] == lockId) {
+                return false;
+            }
+        }
+        if (cacheLine->next != NULL) {
+            entryIndex = 0;
+            cacheLine = cacheLine->next;
+            overflowCacheLineCnt++;
+        } else {
+            break;
+        }
+    }
+
+    // We need another CacheLine
```

Comment:
Move this comment just inside the "then" clause:  you don't need another CacheLine unless the "if" condition is satisfied.